tools/protodoc/protodoc.py

```diff
@@ -194,16 +220,47 @@ def FormatExtension(extension):
     anchor = FormatAnchor('extension_' + extension)
     status = EXTENSION_STATUS_VALUES.get(extension_metadata['status'], '')
     security_posture = EXTENSION_SECURITY_POSTURES[extension_metadata['security_posture']]
+    extension_categories = "\n".join(
+      "  - :ref:`%s <extension_type_%s>`" % (ext, ext)
+      for ext
+      in extension_metadata.get('categories', []))
     return EXTENSION_TEMPLATE.substitute(anchor=anchor,
                                          extension=extension,
                                          status=status,
-                                         security_posture=security_posture)
+                                         security_posture=security_posture,
+                                         categories="%s\n  \n" % extension_categories)
   except KeyError as e:
     sys.stderr.write(
         '\n\nDid you forget to add an entry to source/extensions/extensions_build_config.bzl?\n\n')
     exit(1)  # Raising the error buries the above message in tracebacks.
 
 
+def FormatExtensionType(extension_type):
+  """Format extension metadata as RST.
+
+  Args:
+    extension_type: the name of the extension_type, e.g. com.acme.
+
+  Returns:
+    RST formatted extension type description.
+  """
+  IGNORED_EXTENSIONS = [
+    "envoy.tracers.xray", "envoy.filters.http.cache.simple_http_cache"]
```

Comment:
@htuch 

...also, there were 2 extensions that are in the `extension_db` but dont seem to use `envoy_cc_extension` func

ive added this to ignore them here for now, but not sure if this is the correct resolution

Reply:
Prefer to move them to `envoy_cc_extension` if possible.

Reply:
from further investigation it seems like what was missing wasnt the `envoy_cc_extension` entry after all - but the extension annotation in the relevant proto - ill add these and flag them...